core/ui/ui.go

```diff
@@ -1,17 +1,51 @@
 package ui
 
+import "errors"
+
 // Config is the UI configuration for core.Endpoint
 type Config struct {
 	// HideHostname whether to hide the hostname in the Result
 	HideHostname bool `yaml:"hide-hostname"`
 	// DontResolveFailedConditions whether to resolve failed conditions in the Result for display in the UI
-	DontResolveFailedConditions bool `yaml:"dont-resolve-failed-conditions"`
+	DontResolveFailedConditions bool   `yaml:"dont-resolve-failed-conditions"`
+	Badge                       *Badge `yaml:"badge"`
+}
+
+type Badge struct {
+	ResponseTime *Thresholds `yaml:"response-time"`
+}
+
+type Thresholds struct {
+	Thresholds []int `yaml:"thresholds"`
+}
```

Comment:
Shouldn't this struct be called `ResponseTime` instead?

Reply:
Since there is still an uptime badge that can be configured with thresholds in the same way, I thought that this way would already suffice and therefore prepared it. 
But you could also take out the threshold type directly and replace it with an int[] and thus reduce the config levels.